core/templates/dev/head/pages/exploration_player/AudioBarDirective.js

```diff
@@ -182,6 +182,8 @@ oppia.directive('audioBar', [
 
           $scope.track = {             
             progress: function(progress) {
+              // Update the track progress if argument progress is defined
```

Comment:
This is a little misleading. It implies that nothing is returned if the "argument progress is defined".

Firstly I would change the 'progress' arg to another name. I don't know what its type is. If it's a number then maybe make it progressPercentage to make it clear that it is a number.

Secondly you might consider rewriting as follows (please test this):

```
progress: function(progressPercentage) {
  if (angular.isNumber(progressPercentage)) {
    AudioPlayerService.setProgress(progressPercentage / 100);
  }
  return AudioPlayerService.getProgress() * 100;
}
```

thus removing some of the code branching.

Reply:
Thanks @seanlip! Fixed this.

Reply:
No, it is not fixed. Please read the entire comment I wrote carefully.

Reply:
@seanlip I have removed the code branching and tested it too, it works fine !